### ISSUE TITLE
NPE with Plugin 6.0.1 / Jooq 3.16.1 and 3.15.1

### ISSUE THREAD

**Issue by wulf0r** [etiennestuder/gradle-jooq-plugin#202]
I'm getting an NPE when running the generateJooq task of the Plugin with version 6.0.1:

    Caused by: java.lang.NullPointerException: Cannot invoke "java.net.URL.toExternalForm()" because "schema" is null
      at nu.studer.gradle.jooq.JooqGenerate.writeConfiguration(JooqGenerate.java:243)
      at nu.studer.gradle.jooq.JooqGenerate.generate(JooqGenerate.java:187)

My limited investigation revealed that it's actually the Schema URL which is being passed to SchemaFactory that is null
My debugger says that the method xsdFileName() yields value  "jooq-codegen-3.16.0.xsd" when the NPE happens.

I tried to put this jooq version on the generators CP but that didn't change anything.

Versions:
Win64
JDK 11
Gradle 6.8.1
Plugin id 'nu.studer.jooq' version '6.0.1'
Jooq 3.16.1 and 3.15.1 (regardless of if I set jooq.version)

It's a multi-module project with a custom generator

I can work around the issue by including this buildscript configuration I found in other issues:

    buildscript {
        configurations['classpath'].resolutionStrategy.eachDependency {
            if (requested.group == 'org.jooq') {
                useVersion '3.12.4'
            }
        }
    }

**Comment by ThoSap** [comment]
@etiennestuder I have the same issue with any `3.16.X` version.

The only solution to this issue is to remove the fix of [specify_jooq_config_xml_schema_version/build.gradle](https://github.com/etiennestuder/gradle-jooq-plugin/blob/master/example/specify_jooq_config_xml_schema_version/build.gradle)
```
    configurations['classpath'].resolutionStrategy.eachDependency {
        if (requested.group == 'org.jooq') {
            useVersion "$jooqVersion" // 3.16.2
        }
    }
```
or to explicitly set `useVersion` to `3.15.6`.

**Comment by perlun** [comment]
This should hopefully be fixed by #206. :tada:

**Comment by etiennestuder** [comment]
This should be fixed in 7.0. Note that you have to use jOOQ 3.16.0 or higher to use the gradle-jooq-plugin 7.0.

**Comment by etiennestuder** [comment]
This is fixed in the recently relased v7.0 of the gradle-jooq-plugin. The problems was due to a backward-incompatible change in jOOQ between 3.15 and 3.16.
